### PR TITLE
[FLINK-2526]Add try-catch for task when it stop running

### PR DESCRIPTION
**Pull request by ffbin**
inputProcessor cleanup() may throw IOException.If do not catch it, the next line outputHandler .flushOutputs() will not work and cause output data loss.So i think add some try-catch is necessary.

### CODE REVIEW

File: flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/runtime/tasks/OneInputStreamTask.java [hunk 1]
```diff
@@ -100,9 +100,27 @@ public void invoke() throws Exception {
...
 			inputProcessor.clearBuffers();
-			inputProcessor.cleanup();
-			outputHandler.flushOutputs();
-			clearBuffers();
+
+			try {
+				inputProcessor.cleanup();
+			}
+			catch (Exception e) {
+				LOG.warn("Clean up input processor failed.");
```
Comment (zentol) [inline flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/runtime/tasks/OneInputStreamTask.java]: the exception message shouldn't be discarded.

**Comment by ffbin** [comment]
@zentol Hello! Thank you for your suggestion.I have add exception message in log.

**Comment by StephanEwen** [comment]
The pull request #1017 actually fixes this as well, together with a lot of other things (similar problems are in many places).

Can you have a look at #1017 and see if that would solve your concern?

**Comment by ffbin** [comment]
@StephanEwen  Thanks. I think you pull request has fixes this.I will close my pull request and view your code.

**Pull request closed without merging**